open_discussions/authentication.py

```diff
@@ -14,39 +15,25 @@
 logger = logging.getLogger()
 
 
-def get_encoded_and_signed_subscription_token(user):
-    """
-    Returns a signed and encoded token for subscription authentication_classes
-
-    Args:
-        user (User): user to generate the token for
+class IgnoreExpiredJwtAuthentication(JSONWebTokenAuthentication):
+    """Version of JSONWebTokenAuthentication that ignores JWT values if they're expired"""
 
-    Returns:
-        str: a signed token for subscription authentication_classes
-    """
-    signer = signing.TimestampSigner()
-    return signer.sign(user.username)
+    def get_jwt_value(self, request):
+        """Returns the JWT values as long as it's not expired"""
+        value = super().get_jwt_value(request)
 
+        try:
+            # try to decode the value just to see if it's expired
+            from rest_framework_jwt.settings import api_settings
+            jwt_decode_handler = api_settings.JWT_DECODE_HANDLER
+            jwt_decode_handler(value)
+        except jwt.ExpiredSignature:
+            # if it is expired, treat it as if the user never passed a token
+            return None
+        except:  # pylint: disable=bare-except
+            pass
```

Comment:
What exceptions is this silencing? Should this raise instead if they're unknown?

Reply:
It's silencing all exceptions because the token gets parsed later in the request lifecycle and we want that code to do the error handling of whatever other exceptions there are. We're only concerned with the more specific exception handled above for purposes of ignoring the token.

Reply:
Ok, can you add that as a comment?